### DISCUSSION TITLE
Streaming bluetooth audio from phone-esp32-esp32-bluetooth speaker

### DISCUSSION THREAD

**Discussion by MaximWille** [pschatzmann/arduino-audio-tools#864] [Q&A] [answered]
Hi everyone, 

i'm using two esp32 boards for my project. The aim is to send bluetooth audio to the first esp32 via my iphone. Then the first esp32 connects to a second esp32 through the GPIO Pins. The second esp32 then connects to a bluetooth speaker and sends the music it receives from the first esp32. 

I'm trying to do that by using two examples from the libraries kindly provided by pschatzmann. The first esp32 runs the "bt_music_receiver_simple" example. It just converts the A2DP stream to an i2s signal that is sent to gpio pins as described here: https://github.com/pschatzmann/ESP32-A2DP

The second esp32 runs the "stream-i2s-a2dp" example that can be found here: https://github.com/pschatzmann/arduino-audio-tools 

I'm using the default GPIO Pins. The only thing i changed in the code is obviously the name of the bluetooth speaker. 

The issue is, that the sound received by the speaker ist extremly noisy. In regular intervals, i can hear a glimpse of the song i'm playing back, then it goes back to undefined noise. When i pause the song, the noise stops. That tells me that something is working, but there is an issue with the conversion of the signal at some point. 

Does someone know where my mistake lies?

I have tried converting to a 32bit i2s signal and then back to a 16bit signal for the second esp32 but it doesn't work at all. I'm trying to play music from the spotify app on my iphone. Is A2DP the wrong protocoll to use? It seems like it's the only bluetooth protocoll supported by the library. 

I would be very grateful if someone can help me. Thank you! 
The ultimate goal of my project is to use several esp32 as bluetooth outputs so that several different bluetooth speakers can be connected together to play from one bluetooth source.

**Comment by pschatzmann** [comment, marked as answer]
You should not ignore my [recommendations that can be found in the Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working)

It is quite obvious that you need to systematically search for the source of your error: In can be basically everywhere!

- A2DP needs 16bits on 2 channels, so if you mess around with 32 bits, this will not help.
- Try to find out if the issue is on the output side
- You were not clear if the issue is when you send the audio from your iphone, if so, try to replace it with some other devices
- I don't think that you will have any success in connecting 2 ESP32 over a wire using GPIO pins, whatever you mean with this
- And finally you can also forget about sending audio to multiple receivers via A2DP at the same time: This is not supported by A2DP!
